### PR TITLE
refactor: simplify body timeout

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

File: lib/core/client.js [hunk 1]
```diff
@@ -586,6 +587,14 @@ class Parser extends HTTPParser {
       client[kReset] = true
     }
 
+    if (statusCode >= 200 && request.bodyTimeout) {
+      this.bodyTimeout = setTimeout((self) => {
+        if (!self.paused) {
+          util.destroy(self.socket, new BodyTimeoutError())
+        }
+      }, request.bodyTimeout, this)
```
[the remark this answers is not in the document]
Comment (ronag) [reply lib/core/client.js:595]: There is no closure here? V8 should detect that no?

[the remark this answers is not in the document]
Comment (mcollina) [reply lib/core/client.js:595]: There is still a bit of memory allocated from my experience (at least in some cases).

[the remark this answers is not in the document]
Comment (ronag) [reply lib/core/client.js:595]: I see. I'll move it out.

[the remark this answers is not in the document]
Comment (ronag) [reply lib/core/client.js]: @mcollina fixed PTAL

**Pull request merged**